src/client/containers/PullRequestSummary/BuildSection/Builds/index.js

```diff
@@ -0,0 +1,206 @@
+/* @flow */
+
+import React, { PureComponent } from 'react'
+import Col from 'react-bootstrap/lib/Col'
+import Row from 'react-bootstrap/lib/Row'
+import { getBuilds, getABVBuild } from 'ducks/bfstats/selectors'
+import { buildKatanaBuildLink } from 'routes/helpers'
+import ListGroupItem from 'react-bootstrap/lib/ListGroupItem'
+import moment from 'moment'
+import { BuildsResult } from 'universal/constants'
+import { connect } from 'react-redux'
+
+import { Column, Table } from 'react-virtualized'
+import 'react-virtualized/styles.css'
+import './Builds.css'
+
+export type BuildType = {
+  builder: {
+    name: string,
+    project: string,
+    friendlyName: string
+  },
+  result: number,
+  buildTime: number,
+  finish: string,
+  number: number,
+  start: string,
+}
+
+type PropsType = {
+  repository: string,
+  revision: string,
+  builds: Array<BuildType>,
+  abvBuild: BuildType,
+}
+
+const subHeader = text => (
+  <div className="sub-header">
+    {text}
+  </div>
+)
+
+const successColor = 'green'
+const failureColor = '#e63b72'
+
+const renderBuildTime = (build: BuildType) => {
+  let finish = build.finish
+  if (!finish) {
+    finish = moment(build.start).add(build.buildTime, 's')
+  }
+  const buildDate = moment(finish)
+  return <div style={{ fontSize: '12px' }}>{`${buildDate.format('MMM Do YY, h:mm:ss a')} (${buildDate.fromNow()})`}</div>
+}
+
+const getBuildColor = (result: number) => (result === 0 ? successColor : failureColor)
+
+const renderBuildStatus = (build: BuildType) =>
+  <div style={{ color: getBuildColor(build.result), textTransform: 'uppercase' }}>
+    {BuildsResult[build.result]}
+  </div>
+
+class Builds extends PureComponent {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      expanded: false,
+    }
+  }
+  props: PropsType
+
+  state: {
+    expanded: boolean
+  }
+
+  resultRenderer = ({
+    cellData,
+    columnData,
+    dataKey,
+    isScrolling,
+    rowData,
+    rowIndex,
+  }) => renderBuildStatus(this.props.builds[rowIndex])
```

Comment:
Curious, why write it like this rather than `resultRenderer = row => renderBuildStatus(this.props.builds[row.rowIndex]` ?